third_party/2/concurrent/futures/__init__.pyi

```diff
@@ -8,12 +12,12 @@ class Future(Generic[_T]):
     def running(self) -> bool: ...
     def done(self) -> bool: ...
     def result(self, timeout: float = ...) -> _T: ...
-    def exception(self, timeout: float = ...) -> BaseException: ...
+    def exception(self, timeout: float = ...) -> _ExceptionType: ...
```

Comment:
Don't return Unions, just "Any".

See, the way Mypy treats those cases is a little unintuitive:
- a function returning `Union[A, B]` can return either type A or type B; so
- code using that return value is only safe if it uses the value in places safe for both types; for example
- using the return value's methods and attributes is only okay if they are present in all types within the union (everything else will raise an error).

In our case the union is with any old-style class. So it means only methods and attributes present on both old-style classes and BaseExceptions are acceptable.